### DISCUSSION TITLE
How to keep dojo alive even when a DojoCrashError is raised

### DISCUSSION THREAD

**Discussion by wclsn** [lean-dojo/LeanDojo#150] [Q&A] [answered]
Hello everyone,

I have an unusual question to discuss. I've noticed that when a DojoCrashError occurs, the Dojo environment collapses, requiring us to reinitialize it in order to restore normal functionality. 

However, this process of reinitializing the Dojo environment can be time-consuming and memory-intensive. My question is, is it possible to ensure the continuous operation of the Dojo environment and return to the previous state even when a DojoCrashError is raised? Would it be necessary to make modifications to specific sections of the Dojo source code in order to achieve this?

**Comment by yangky11** [comment, marked as answer]
Unfortunately, I don't see a good way to achieve this without significant modification of the code. DojoCrashError happens often because of non-recoverable errors from Lean. The best way is probably recording all necessary steps before it crashes and trying to replay them with a new Dojo instance.

**Reply by wclsn** [reply]
I also think it is a little hard. Whatever, thank you for your reply.
